crates/sui-storage/src/event_store/sql.rs

```diff
@@ -115,15 +116,43 @@ impl SqlEventStore {
             .journal_mode(SqliteJournalMode::Wal)
             // Normal vs Full sync mode also speeds up writes
             .synchronous(SqliteSynchronous::Normal)
+            // Minimal journal size and frequent autocheckpoints help prevent giant WALs
+            .pragma("journal_size_limit", "0")
+            .pragma("wal_autocheckpoint", "400") // In pages of 4KB each
             .create_if_missing(true);
         options.log_statements(log::LevelFilter::Off);
         let pool = SqlitePool::connect_with(options)
             .await
             .map_err(convert_sqlx_err)?;
         info!(?db_path, "Created/opened SQLite EventStore on disk");
+
         Ok(Self { pool })
     }
 
+    /// Starts a WAL truncation/cleanup periodic task at interval duration
+    pub async fn wal_cleanup_thread(&self, wal_cleanup_interval: Option<Duration>) {
+        if let Some(cleanup_interval) = wal_cleanup_interval {
+            let mut interval = tokio::time::interval(cleanup_interval);
+            loop {
+                interval.tick().await;
+                info!("Running SQLite WAL truncation...");
+                let _ = self.force_wal_truncation().await.map_err(|e| {
+                    warn!("Unable to truncate Event Store SQLite WAL: {}", e);
+                });
+            }
+        }
+    }
+
+    /// Force the SQLite WAL to be truncated.  This mighyt be occasionally necessary if somehow the WAL
+    /// grows too big.
+    pub async fn force_wal_truncation(&self) -> Result<(), SuiError> {
+        self.pool
+            .execute("PRAGMA wal_checkpoint(TRUNCATE)")
+            .await
+            .map_err(convert_sqlx_err)?;
```

Comment:
I'll add span instrumentation so we can track the length.   Does it block other ops - it might.   I would argue that for things that block you want to do them as frequently as possible because then each task is very small and has less of an impact.  If you do it too infrequently then it might take a lot more work.   That said, truncating a multi >40GB WAL file was maybe 2 seconds, so if we run it every 5 min it should be really really quick.